bin/reth/src/node/mod.rs

```diff
@@ -710,11 +728,13 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
             .set_head(head)
             .listener_addr(SocketAddr::V4(SocketAddrV4::new(
                 Ipv4Addr::UNSPECIFIED,
-                self.network.port.unwrap_or(DEFAULT_DISCOVERY_PORT),
+                // set discovery port based on instance number
+                self.network.port.unwrap_or(DEFAULT_DISCOVERY_PORT + self.instance - 1),
```

Comment:
The behavior of the other ports is that if you specify a port, e.g. for RPC, that is used as the starting port (i.e. the port for instance 1). I would like to have this behavior for the network port too.

In the current implementation, if I set the network port to something else and started multiple instances, they would all use the same port.